ROMUtils.cpp

```diff
@@ -55,11 +55,75 @@ namespace ROMUtils
     /// A pointer into the ROM data to start reading from.
     /// </param>
     /// <param name="outputSize">
-    /// The predicted size of the output data.
+    /// The predicted size of the output data.(unit: Byte)
     /// </param>
     /// <return>A pointer to decompressed data.</return>
-    unsigned char *RLEDecompress(unsigned char *data, int outputSize)
+    unsigned char *RLEDecompress(unsigned char *data, int address, int outputSize)
     {
-        // TODO
+        unsigned char *OutputLayerData = new unsigned char[outputSize];
+        unsigned char *dst;
+        int src = address;
+        int ctrl;
+        int nn;
+
+        for(int ii=0; ii<2; ii++)
+        {
+            dst = &OutputLayerData[ii];
+            if(data[src++] == 1)
+            {
+                while(1)
```

Comment:
If we use while(1) and then only break at the end of data, then memory could be could get corrupted if we call this function on the wrong address. I think it would be better if the function tracks the size of what it is decompressing, and if it does not match outputsize (or tries to write more than outputsize), it deletes OutputLayerData and returns nullptr, so we can give an error dialog instead of just crashing.

Reply:
I will add codes for this inside that loop.